components/engine_rocks/src/config.rs

```diff
@@ -334,31 +354,80 @@ macro_rules! numeric_enum_mod {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum CompactionPriority {
+    ByCompensatedSize = 0,
+    OldestLargestSeqFirst = 1,
+    OldestSmallestSeqFirst = 2,
+    MinOverlappingRatio = 3,
+}
```

Comment:
And rename `numeric_enum_mod` to something like `impl_enum_mirror_type`

Reply:
> Can put these inside macro too 

You mean we can put the definition of enums like `CompactionPriority ` inside the macro `numeric_enum_mod`

>  And rename numeric_enum_mod to something like impl_enum_mirror_type

Reasonable. Since this enum not only support numberic value when deserialization, we need to change it to a more proper name